K2Bridge/Factories/BucketFactory.cs

```diff
@@ -119,5 +119,29 @@ public static RangeBucket CreateRangeBucket(string primaryKey, DataRow row, ILog
 
             return rb;
         }
+
+        /// <summary>
+        /// Create a new <see cref="HistogramBucket" from a given <see cref="DataRow"/>/>.
+        /// </summary>
+        /// <param name="row">The row to be transformed to bucket.</param>
+        /// <returns>A new HistogramBucket.</returns>
+        public static HistogramBucket CreateHistogramBucket(string primaryKey, DataRow row, ILogger logger)
+        {
+            Ensure.IsNotNull(row, nameof(row));
+
+            var key = row[0];
+            var count = row[BucketColumnNames.Count];
+            var keyed = row.Table.Columns[0].ToString().Split('%')[1];
+
+            var hb = new HistogramBucket
+            {
+                DocCount = Convert.ToInt32(count),
+                Key = Convert.ToDouble(key),
```

Comment:
Key can be DBNull, we should handle that

Reply:
also `timestamp` in datehistogram

Reply:
Done

Reply:
It can't be actual `null`, it will give you an object of the type "DBNull"

Reply:
was this addressed?

Reply:
yes done